pkg/logs/input/file/scanner.go

```diff
@@ -49,20 +50,25 @@ type Scanner struct {
 	registry            auditor.Registry
 	tailerSleepDuration time.Duration
 	stop                chan struct{}
+	// set to true if we want to use `ContainersLogsDir` to validate that a new
+	// pod log file is being attached to the correct containerID.
+	// Feature flag defaulting to false, use `logs_config.validate_pod_container_id`.
+	validatePodContainerID bool
 }
 
 // NewScanner returns a new scanner.
 func NewScanner(sources *config.LogSources, tailingLimit int, pipelineProvider pipeline.Provider, registry auditor.Registry, tailerSleepDuration time.Duration) *Scanner {
 	return &Scanner{
-		pipelineProvider:    pipelineProvider,
-		tailingLimit:        tailingLimit,
-		addedSources:        sources.GetAddedForType(config.FileType),
-		removedSources:      sources.GetRemovedForType(config.FileType),
-		fileProvider:        NewProvider(tailingLimit),
-		tailers:             make(map[string]*Tailer),
-		registry:            registry,
-		tailerSleepDuration: tailerSleepDuration,
-		stop:                make(chan struct{}),
+		pipelineProvider:       pipelineProvider,
+		tailingLimit:           tailingLimit,
+		addedSources:           sources.GetAddedForType(config.FileType),
+		removedSources:         sources.GetRemovedForType(config.FileType),
+		fileProvider:           NewProvider(tailingLimit),
+		tailers:                make(map[string]*Tailer),
+		registry:               registry,
+		tailerSleepDuration:    tailerSleepDuration,
+		stop:                   make(chan struct{}),
+		validatePodContainerID: coreConfig.Datadog.GetBool("logs_config.validate_pod_container_id"),
```

Comment:
Passing the option's value to `NewScanner` instead of calling the config package would be more consistent with many logs agent packages.

If this is a temporary config option that will disappear, I don't mind that much though.

Reply:
Agreed.